File: app/src/main/java/com/example/cs_topics_project_test/function_java/TaskOutline.java
package com.example.cs_topics_project_test.function_java;

public interface TaskOutline {
    String getTaskName(); // the name of the task
    Date getDueDate(); // the due date
    Time getDueTime(); // the due time
    boolean isTaskCompleted(); // returns whether the task is complete
    String getTaskDescription(); // returns task description

    // Modifiers
    void setTaskName(String name); // resets the task name
    void setDueDate(Date date); // changes due date
    void setDueTime(Time time); // changes the time that tasks was due
    void taskCompleted(); // sets task to completed
    void taskNotCompleted(); // sets task to incomplete
    void setTaskDescription(String description); // resets the task description
}

File: app/src/main/java/com/example/cs_topics_project_test/function/TaskOutline.kt
package com.example.cs_topics_project_test.function

interface TaskOutline {
    fun getTaskName(): String // the name of the task
    fun getDueDate(): Date // the due date
    fun getDueTime(): Time // the due time
    fun isTaskCompleted(): Boolean // returns whether the task is complete
    fun getTaskDescription(): String // returns task description

    // Modifiers
    fun setTaskName(name: String) // resets the task name
    fun setDueDate(date: Date) // changes due date
    fun setDueTime(time: Time) // changes the time that tasks was due
    fun taskCompleted() // sets task to completed
    fun taskNotCompleted() // sets task to incomplete
    fun setTaskDescription(description: String) // resets the task description
}